src/Storages/StorageBuffer.cpp

```diff
@@ -1197,6 +1197,28 @@ void StorageBuffer::alter(const AlterCommands & params, ContextPtr local_context
     setInMemoryMetadata(new_metadata);
 }
 
+UInt64 checkUnderflowAndGetUInt64(const ASTPtr & arg, const String & arg_name)
+{
+    /**
+      * Do not force UInt64 type for args, it'll backward incampatible,
+      * there are exponential notation usages for shortness.
+      */
+    const auto & value = arg->as<ASTLiteral &>().value;
+    if (value.getType() > Field::Types::UInt64)
```

Comment:
I am not sure whether it is correct to compare `Types::Which` in such way, it must just compare enum value numbers which are not comparable by size